compnerd/gha-setup-swift
# Enable usage of a custom Swift toolchain

A custom Swift toolchain produced as a Github release can be downloaded and installed instead of the official Swift.org toolchains. This allows for a project to have a faster iteration time if they need to use the latest code or a forked toolchain.

## Tests
* Ensure that it works with the swift.org official builds (no regressions), by using branch and tag arguments and validating the version of the toolchain that is downloaded. Added a action to test it on all OSes.
* Ensure that it works with a release in a (private) repository forked from compnerd/swift-build
